ccm_web/client/src/pages/GradebookCanvas.tsx

```diff
@@ -0,0 +1,395 @@
+    parse<GradebookRecord>(file, {
+      header: true,
+      skipEmptyLines: true,
+      complete: function (results) {
+        handleParseComplete(results)
+      }
+
+    })
+  }
+
+  const resetPageState = (): void => {
+    setPageState({ state: GradebookCanvasPageState.Upload })
+  }
+
+  const getOutputFilename = (file: File | undefined): string => {
+    if (file === undefined) return ''
+    const splitName = file.name.split('.')
+    const filenameIndex = splitName.length >= 2 ? splitName.length - 2 : 0
+    splitName[filenameIndex] = splitName[filenameIndex] + '-geff'
+    return splitName.join('.')
+  }
+
+  const setCSVtoDownload = (data: GradebookRecord[]): void => {
+    let csvContent = 'data:text/csv;charset=utf-8,'
+    data.forEach(function (record, index) {
+      csvContent += record['SIS Login ID'] + ',' + record['Final Grade'] + (index < data.length ? '\n' : '')
+    })
+    setDownloadData({ data: encodeURI(csvContent), fileName: getOutputFilename(file) })
+  }
+
+  const handleNoLetterGradesError = (): void => {
+    setPageState({ state: GradebookCanvasPageState.InvalidUpload, errorMessage: [<Typography key='0'><Link href='#'>Grading Scheme in settings</Link> needs to be check marked for letter grade to appear in the CSV file.</Typography>, <Typography key='1'>{renderUploadAgainButton()}</Typography>] })
+  }
+
+  const handleRowLevelInvalidationError = (errorMessage: JSX.Element[], invalidations: GradebookRowInvalidation[]): void => {
+    setPageState({ state: GradebookCanvasPageState.InvalidUpload, invalidations: invalidations, errorMessage: errorMessage })
+  }
+
+  const handleParseSuccess = (grades: GradebookRecord[]): void => {
+    setPageState({ state: GradebookCanvasPageState.Confirm, grades: grades })
+    setCSVtoDownload(grades)
+  }
+
+  const handleParseComplete = (results: ParseResult<GradebookRecord>): void => {
+    const data = results.data.slice(1) // The first row is possible scores
+
+    if (data[0]['Final Grade'] === undefined) {
+      handleNoLetterGradesError()
+      return
+    }
+
+    let invalidations: GradebookRowInvalidation[] = []
+
+    const gradeMismatchValidator = new CurrentAndFinalGradeMatchGradebookValidator()
+
+    data.forEach(record => {
+      invalidations = invalidations.concat(gradeMismatchValidator.validate(record, data.indexOf(record) + 1))
+    })
+
+    if (invalidations.length > 0) {
+      handleRowLevelInvalidationError([<div key='0'>There are blank cells in the gradebook. Please enter 0 or EX (for excused) for any blank cells in the gradebook and export a new CSV file.</div>], invalidations)
+    } else {
+      handleParseSuccess(data)
+    }
+  }
+
+  const renderUploadHeader = (): JSX.Element => {
+    return <span>
+      <Typography variant='h6'>Upload your CSV File</Typography>
+      <Typography>The CSV file will be formatted by trimming out nonessential columns.</Typography>
+      <br/>
```

Comment:
I feel there should be some gap between Page title which is Canvas Gradebook Converter and the text below starting with `Upload your....`
![Screen Shot 2021-05-06 at 7 26 14 AM](https://user-images.githubusercontent.com/8579775/117291356-e3ea8380-ae3c-11eb-9856-4e694867c0c1.png)

Reply:
I agree. Here is an example: 
<img width="467" alt="Screen Shot 2021-05-06 at 8 45 47 AM" src="https://user-images.githubusercontent.com/46663379/117300620-be16ac00-ae47-11eb-88f1-bd270379ee78.png">